Formula/scotch.rb

```diff
@@ -20,24 +21,32 @@ class Scotch < Formula
     sha256 cellar: :any_skip_relocation, x86_64_linux:  "f93fe6e8bcdb31271f7d082915061d6fa2b9e415611a6f52929dbb45a48dd847"
   end
 
-  depends_on "open-mpi" => :build
+  depends_on "open-mpi"
 
   uses_from_macos "bison"
   uses_from_macos "flex"
   uses_from_macos "zlib"
 
   def install
-    cd "src"
```

Comment:
Let's keep this line so that we don't have to indent everything inside the `install` method.

Reply:
I changed it because I saw this warning
```
/System/Library/Frameworks/Ruby.framework/Versions/2.6/usr/lib/ruby/2.6.0/fileutils.rb:128: warning: conflicting chdir during another chdir block
```
Is that OK?

Reply:
Do you still see the warning after getting rid of the `chdir` call?

Reply:
I'm confused here . If I use `cd`, I'll get the warning. But there's actually no `chdir` in the script.
If I use `chdir "src" do` the warning will go off.

Reply:
Can you show the exact form of the `install` method using `cd` that shows the warning? It's hard to say what's happening when I'm not sure what you're talking about.

Reply:
Here's the warning message if we use `cd "src"`. 
```
brew install --build-from-source scotch.rb
Error: Failed to load cask: scotch.rb
Cask 'scotch' is unreadable: wrong constant name #<Class:0x00007fb9485639a8>
Warning: Treating scotch.rb as a formula.
==> Downloading https://gitlab.inria.fr/scotch/scotch/-/archive/v6.1.2/scotch-v6
Already downloaded: /Users/gux215/Library/Caches/Homebrew/downloads/c9a998359af7ef4bd5f5c78f45faa52e0b5e8475cd28b9ff6cd8ad31c811d0ad--scotch-v6.1.2.tar.bz2
/System/Library/Frameworks/Ruby.framework/Versions/2.6/usr/lib/ruby/2.6.0/fileutils.rb:128: warning: conflicting chdir during another chdir block
==> make scotch ptscotch
==> make prefix=/usr/local/Cellar/scotch/6.1.2 install
🍺  /usr/local/Cellar/scotch/6.1.2: 76 files, 5.7MB, built in 11 seconds
```
I searched other formulae and most of them use
```
cd "src" do

end
```
which won't show `warning: conflicting chdir during another chdir block`

Reply:
Please copy and paste the contents of the `install` method that produces that warning here. Is it the one already in your PR?

Reply:
I hope I'm making it clear this time:
The current code:
```
  def install
    cd "src"
    (buildpath/"src").install_symlink "Make.inc/Makefile.inc.i686_mac_darwin10" => "Makefile.inc"
    system "make"
    system "make", "prefix=#{prefix}", "install"
  end
```
Or the one that I modified
```
  def install
    cd "src"
    (buildpath/"src").install_symlink "Make.inc/Makefile.inc.i686_mac_darwin10" => "Makefile.inc"

    inreplace "Makefile.inc" do |s|
      s.change_make_var! "CCS", ENV.cc
      s.change_make_var! "CCP", "mpicc"
      s.change_make_var! "CCD", "mpicc"
    end

    system "make", "scotch", "ptscotch"
    system "make", "prefix=#{prefix}", "install"

    pkgshare.install "check/test_strat_seq.c"
    pkgshare.install "check/test_strat_par.c"
  end
```
Both of them show the warning `conflicting chdir during another chdir block`
If I change it to
```
  def install
    cd "src" do
      (buildpath/"src").install_symlink "Make.inc/Makefile.inc.i686_mac_darwin10" => "Makefile.inc"

      inreplace "Makefile.inc" do |s|
        s.change_make_var! "CCS", ENV.cc
        s.change_make_var! "CCP", "mpicc"
        s.change_make_var! "CCD", "mpicc"
      end

      system "make", "scotch", "ptscotch"
      system "make", "prefix=#{prefix}", "install"

      pkgshare.install "check/test_strat_seq.c"
      pkgshare.install "check/test_strat_par.c"
    end
  end
```
There's no warning.

Reply:
That's much clearer; thank you. Let's put everything inside the `cd "src" do` block then.